### PR TITLE
tighten the isUrlEncoded check

### PR DESCRIPTION
**Pull request by kahing**
ideally we shouldn't need this function and instead never double
encode strings, but auditing for that is beyond what I have time
for. currently, putBlob(" ") and putBlob("%20") behave the same
way which is arguably incorrect

### CODE REVIEW

**Comment by andrewgaul** [comment]
More reports of URL encoding issues at [JCLOUDS-217](https://issues.apache.org/jira/browse/JCLOUDS-217).

File: core/src/main/java/org/jclouds/util/Strings2.java [hunk 1]
```diff
@@ -90,11 +90,34 @@ public String load(Character plain) throws ExecutionException {
...
+
+      // ensure that all % are followed by 2 hexadecimal characters
+      int percentIdx = 0;
+      while ((percentIdx = in.indexOf('%', percentIdx)) != -1) {
+         if (percentIdx + 2 >= in.length()) {
+            return false;
+         }
+         if (!isHexadecimal(in.charAt(percentIdx + 1)) ||
+                 !isHexadecimal(in.charAt(percentIdx + 2))) {
+            System.err.println("% is followed by non-alphanumeric " + in.substring(percentIdx, percentIdx + 2));
```
Comment (andrewgaul) [inline core/src/main/java/org/jclouds/util/Strings2.java]: Remove -- we cannot spam `System.err`.  Optionally introduce a logger instead.

**Comment by andrewgaul** [comment]
Pushed to master as 82ab88d5896a6628f3fbb42a3c4bce39ce843fa1.

**Comment by andrewgaul** [comment]
This commit causes failures in core for me:

```
  BindMapToStringPayloadTest.testDecodes:76 expected [{"changePassword":{"adminPass":"foo"}}] but found [%7B"changePassword":%7B"adminPass":"foo"%7D%7D]
  HttpRequestTest.testAddBase64AndUrlEncodedQueryParams:78 expected [GET http://goo.com:443?header1=ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789%2B/%3D&header2=ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789%2B/%3D HTTP/1.1] but found [GET http://goo.com:443?header1=ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789%20/%3D&header2=ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/= HTTP/1.1]
  HttpRequestTest.testEncodesOnlyOnce:55 expected [{method=GET, endpoint=http://goo.com:443?header=hello%3F, headers={}}] but found [{method=GET, endpoint=http://goo.com:443?header=hello?, headers={}}]
  RestAnnotationProcessorTest.testCreateGetRequest:2133 expected [/qu%3Fstion] but found [/qu]
```

**Comment by andrewgaul** [comment]
Reverted in c22afc5a55db1539d2e796f7244f9d2b3da57b5a.

**Comment by kahing** [comment]
giving up, the code is buggy but there are many dependencies on its buggy behaviors

**Pull request closed without merging**